src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```diff
@@ -247,28 +247,22 @@ class LdapAccessPolicyDAO(
         ldapSearchStream(resourceDn(resource), SearchScope.SUB, Filter.createEqualityFilter("objectclass", ObjectClass.policy))(unmarshalAccessPolicy)
       ))
 
-  override def listAccessPoliciesForUser(resource: FullyQualifiedResourceId, user: WorkbenchUserId): IO[Set[AccessPolicy]] =
+
+
+  override def listAccessPoliciesForUser(resource: FullyQualifiedResourceId, userId: WorkbenchUserId): IO[Set[AccessPolicy]] =
     for {
-      memberOfs <- ldapLoadMemberOf(user)
-      accessPolicies <- {
-        val fullyQualifiedPolicyIds = memberOfs.toList.mapFilter { str =>
-          for {
-            subject <- Either.catchNonFatal(dnToSubject(str)).toOption
-            fullyQualifiedPolicyId <- subject match {
-              case sub: FullyQualifiedPolicyId
-                  if sub.resource.resourceId == resource.resourceId && sub.resource.resourceTypeName == resource.resourceTypeName =>
-                Some(sub)
-              case _ => None
-            }
-          } yield fullyQualifiedPolicyId
-        }
-        val filters = fullyQualifiedPolicyIds
-          .grouped(batchSize)
-          .map(batch => Filter.createORFilter(batch.map(r => Filter.createEqualityFilter(Attr.policy, r.accessPolicyName.value)).asJava))
-          .toSeq
-        executeLdap(IO(ldapSearchStream(resourceDn(resource), SearchScope.SUB, filters: _*)(unmarshalAccessPolicy).toSet))
-      }
-    } yield accessPolicies
+      allPolicies <- listAccessPolicies(resource)
+      userPolicies <- allPolicies.filterA(p => isUserMemberOfPolicy(p, userId))
+    } yield userPolicies.toSet
+
+
+  private def isUserMemberOfPolicy(policy: AccessPolicy, userId: WorkbenchUserId): IO[Boolean] = {
+    if (policy.members.contains(userId)) {
+      IO.pure(true)
+    } else {
+      policy.members.collect{case x:WorkbenchGroupName => x}.toList.existsM(ldapIsUserMemberOfGroup(_, userId))
```

Comment:
`members.collect{case x:WorkbenchGroupName => x}.toList.existsM(ldapIsUserMemberOfGroup(_, userId))` is repeated in `ldapIsUserMemberOfGroup`. Can you pull that into a function named like `isSubGroupMember`?

Reply:
this moved, but I made the change in the new home LdapSupport